src/node/chainstate.h

```diff
@@ -25,6 +25,7 @@ struct ChainstateLoadOptions {
     bool reindex{false};
     bool reindex_chainstate{false};
     bool prune{false};
+    bool fail_on_insufficient_dbcache{false};
```

Comment:
In commit "init, validation: Improve handling if VerifyDB() fails due to insufficient dbcache" (cf307244db97e7b9e29a3d0ff948a6fa7877b548)

`true` might be a safer default for libbitcoinkernel. Also the name of this option `ChainstateLoadOptions::fail_on_insufficient_dbcache` seems a little misleading because it seems like it is referring to insuffcient dbcache for loading, not insufficient_dbcache for verification. Would maybe change it to `ChainstateLoadOptions::require_full_verification` with a comment that setting the option to false can allow using a smaller dbcache size.

Reply:
changed to `ChainstateLoadOptions::fail_on_insufficient_dbcache`

I don't understand the comment. It's not an option the user can set, it's determined by `-checkblocks` and `-checklevel`. So if a user wants to run with an extremely low dbcache, they should change those parameters (or just use the default so that init will only warn).

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25574#discussion_r1109110569

Thanks for implementing the suggestion. This comment was not about `bitcoind`, but about code that uses libbitcoinkernel such as:

https://github.com/bitcoin/bitcoin/blob/fe1b3256888bd0e70d0c9655f565e139ec87b606/src/bitcoin-chainstate.cpp#L93-L96

Code like this has it's own options and defaults, and is not aware of `checkblocks` or `checklevel`. It is just calling and using the `LoadChainstate` function directly. From the perspective of a `LoadChainstate` caller, I think `fail_on_insufficient_dbcache` could have been misleading because it sounds like it would allow loading with an insufficient dbcache, not just verifying with an insufficient dbcache. I do think in either case it wouldn't be bad for the option to have a code comment, so I suggested one below.